rest/client/request.go

```diff
@@ -198,6 +200,12 @@ func (c *communicatorImpl) retryRequest(ctx context.Context, info requestInfo, d
 				return resp, nil
 			} else if resp.StatusCode == http.StatusConflict {
 				return nil, HTTPConflictError
+			} else if resp.StatusCode == http.StatusBadRequest {
```

Comment:
all 400 status codes are considered unrecoverable, so let's just return for any 4xx code.

Also this is probably a good time to mention rehttp, the library with an http client that deals with retry logic, and we can replace this code with it?